src/boot/index.js

```diff
@@ -17,13 +17,44 @@ import { isBrowserSupported } from './browser-check';
 
 if (isBrowserSupported()) {
   const settings = parseJsonConfig(document);
+  // Use the asset root and sidebar app locations specified in the host page,
+  // if present. This is used by the Hypothesis browser extensions to make the
+  // boot script load assets bundled with the extension.
+  let assetRoot;
+  if (settings.assetRoot) {
+    // The `assetRoot` setting is assumed to point at the root of the contents of
+    // the npm package.
+    assetRoot = settings.assetRoot + 'build/';
+  }
+  let sidebarAppUrl = settings.sidebarAppUrl;
+
+  // Otherwise, try to determine the default root URL for assets and the sidebar
+  // application from the location where the boot script is hosted.
+  try {
+    const script = /** @type {HTMLScriptElement} */ (document.currentScript);
+    let scriptUrl = new URL(script.src);
+
+    // We only use the bundled sidebar HTML and assets if the boot script has
+    // its original name. If the `<script>` tag references a custom name
+    // (eg. as in "https://hypothes.is/embed.js") then we skip this and fall
+    // back to the URLs embedded in the boot script.
+    if (scriptUrl.pathname.endsWith('/boot.js')) {
```

Comment:
I don't propose a better solution, but this does feel fragile and hard to understand.  I wonder if there is something that can be driven from the config itself to trigger this condition to make it feel a bit more explicit.